### PR TITLE
Exosuit Medical Beamgun

### PR DESCRIPTION
**Pull request by GunHog**
Adds the Exosuit Medical Beamgun!
- Odysseus version of the Medical Beamgun
- Medical mech equipment is now under /medical/ to prevent copypaste
- Changed some of the medbeam gun code to support the mounted version
- Sprites by @WJohn!
- <b>New!:</b> New Medical beam sprite!

:cl: Gun Hog
rscadd: "Nanotrasen has drafted a design for an exosuit version of the medical nanite projector normally issued to its Emergency Response Personnel. With sufficient research, a prototype may be fabricated for field testing!"
/:cl:

screenshot 2016-01-25 20 10 03

### CODE REVIEW

**Comment by RemieRichards** [comment]
@GunHog for the future, it's /:cl: not :/cl:, I've edited it for you.

**Comment by GunHog** [comment]
Why must you touch me in such ways? Thanks for correction!

**Comment by Fox-McCloud** [comment]
This is completely overpowered---and there was already a discussion about how science doesn't need yet another toy that pretty much invalidates a department

**Comment by duncathan** [comment]
Science != robotics
Roboticists will actually make medbay a mech, from my experience

**Comment by Fox-McCloud** [comment]
Science does equal robotics though; they can build mech consoles for themselves at 5-10 minutes in.

**Comment by MMMiracles** [comment]
Yeah but now theres a reason to build more mechs other than combat now.

**Comment by Fox-McCloud** [comment]
IMO, a better way to do that would be to make the syringe gun not shit (ie: produce syringes internally rather than have to be manually loaded, one at a time).

**Comment by GunHog** [comment]
>IMO, a better way to do that would be to make the syringe gun not shit (ie: produce syringes internally rather than have to be manually loaded, one at a time).

That is not a bad idea, especially since you can make them from autolathes now. I will save that for another PR!

Keep in mind that the Bio 6 gate will ensure that the sleeper and syringe gun are not rendered obsolete - Oddys can be made at roundstart, the syringe gun requires minor research, but you need a bit of work and help to reach Bio 6.

File: code/game/mecha/equipment/tools/medical_tools.dm [hunk 1]
```diff
@@ -1,7 +1,36 @@
...
 
-/obj/item/mecha_parts/mecha_equipment/sleeper
+/obj/item/mecha_parts/mecha_equipment/medical/New()
+	..()
+	SSobj.processing |= src
+
+
+/obj/item/mecha_parts/mecha_equipment/medical/can_attach(obj/mecha/medical/M)
+	if(..())
+		if(istype(M))
```
Comment (neersighted) [inline code/game/mecha/equipment/tools/medical_tools.dm]:
```
if(..() && istype(M))
```

File: code/game/mecha/equipment/tools/medical_tools.dm [hunk 2]
```diff
@@ -1,7 +1,36 @@
...
+			return 1
+	return 0
+
+/obj/item/mecha_parts/mecha_equipment/medical/attach(obj/mecha/M)
+	..()
+	SSobj.processing |= src
```
Comment (neersighted) [inline code/game/mecha/equipment/tools/medical_tools.dm]: Isn't this already added to the processing list in `New()`?

File: code/game/mecha/equipment/tools/medical_tools.dm [hunk 3]
```diff
@@ -1,7 +1,36 @@
...
+
+/obj/item/mecha_parts/mecha_equipment/medical/Destroy()
+	SSobj.processing.Remove(src)
```
Comment (neersighted) [inline code/game/mecha/equipment/tools/medical_tools.dm]: `-=`

File: code/game/mecha/equipment/tools/medical_tools.dm [hunk 4]
```diff
@@ -1,7 +1,36 @@
...
+	return ..()
+
+/obj/item/mecha_parts/mecha_equipment/medical/process()
+	if(!chassis)
+		SSobj.processing.Remove(src)
```
Comment (neersighted) [inline code/game/mecha/equipment/tools/medical_tools.dm]: `-=`

File: code/game/mecha/equipment/tools/medical_tools.dm [hunk 5]
```diff
@@ -516,3 +532,33 @@
...
+	energy_drain = 10
+	range = MELEE|RANGED
+	equip_cooldown = 0
+	origin_tech = "combat=5;materials=6;powerstorage=7;biotech=6"
+	var/obj/item/weapon/gun/medbeam/mech/medigun
+	materials = list(MAT_METAL = 15000, MAT_GLASS = 8000, MAT_PLASMA = 3000, MAT_GOLD = 8000, MAT_DIAMOND = 2000)
+
+/obj/item/mecha_parts/mecha_equipment/medical/mechmedbeam/New()
+	..()
+	medigun = new /obj/item/weapon/gun/medbeam/mech(src)
```
Comment (neersighted) [inline code/game/mecha/equipment/tools/medical_tools.dm]: `medigun = new(src)`

File: code/modules/projectiles/guns/medbeam.dm [hunk 6]
```diff
@@ -58,24 +61,29 @@
 
 	if(world.time <= last_check+check_delay)
 		return
-	
+
 	last_check = world.time
 
-	if(get_dist(H,current_target)>max_range || !los_check(H,current_target))
+	if(get_dist(source,current_target)>max_range || !los_check(source,current_target))
```
Comment (neersighted) [inline code/modules/projectiles/guns/medbeam.dm]: Spaces

File: code/modules/projectiles/guns/medbeam.dm [hunk 7]
```diff
@@ -97,10 +105,18 @@
...
 /obj/effect/ebeam/medical
 	name = "medical beam"
+
+//////////////////////////////Mech Version///////////////////////////////
+/obj/item/weapon/gun/medbeam/mech
+	mounted = 1
+
+/obj/item/weapon/gun/medbeam/mech/New()
+	..()
+	SSobj.processing.Remove(src) //Mech mediguns do not process until installed, and are controlled by the holder obj
```
Comment (neersighted) [inline code/modules/projectiles/guns/medbeam.dm]: `-=`

**Comment by newfren** [comment]
:+1: Odysseus buffs are hardly science buffs given how bad they are currently.

**Comment by kevinz000** [comment]
wheres the tablecuffing?

**Comment by GunHog** [comment]
@kevinz000 Odysseus mechs are already mobile tables, Mekhi! I will be happy to morphine you with the syringe gun and hold you in the sleeper!

**Comment by AdamElTablawy** [comment]
This requires bio 6, which is absurdly hard to get short of an alien brain.

It also requires mining to co-operate.

It also requires someone competent to be doing R&D and be printing an oddyseus.

The chances of this happening are low enough to barely occur once a week. The chances of someone using it are even lower.

Not only that, but almost everything available in science at higher levels is ridiculously overpowered, and science can already invalidate entire departments if they're bored just through making public autocloners and public sleepers/cyrotubes.

To be frank, while nobody in their right mind would call this balanced, it's certainly a fair reward for managing to get these ridiculous levels with ridiculous conditions in rounds that are liable to last 30 minutes, tops, before science gets bombed/a nukeop pod farts right into R&D and shoots your ass.

**Comment by GunHog** [comment]
New medbeam sprite by @WJohn! Sadly, I could not properly make a gif without the program bugging out...

**Comment by GunHog** [comment]
Should be review-able now.

**Comment by tkdrg** [comment]
Not everything needs to be in R&D, specially not antag-only OP items. I know you buff science at every possible opportunity but come on, making medical completely useless is ridiculous. Also, bio 6 isn't that hard to get at all, sentience potions are bio 5 and so are xrays, so all you need is either mining or xenobio. Printing an odysseus is also trivial. All you really need are a little bit of minerals and a decent player in R&D who wants this. And @duncathan of course science won't make medbay a mech; they'll make it for themselves and then proceed to make medical useless once more.

**Comment by GunHog** [comment]
Do not be ridiculous, @tkdrg. It takes a specific set of conditions for Science to achieve top end levels: Miners must have survived and brought back minerals. There must be a competent R&D player as well as a Xenobiologist who is willing to share a potion. The antagonst player(s) must not have prevented progression of any of these areas. This is most likely to happen under strong pressure from a large threat (such as a blob) or an unusually long( for /tg/ players) round.


The origin tech of an Xray gun is `origin_tech = "combat=5;materials=3;magnets=2;syndicate=2"`.

I think that most Roboticists prefer a _combat_ mech to a healing one. Besides, in my observations, the reason people outside Science do not get items is because _they rarely request them._ I often offer guns and implants to Security. They rarely send an officer to pick them up. Cargo never requests a Ripley. Engineers rarely drop by to pick up NVG mesons or welders. The people who come by to Science that I see are miners requesting better gear, the janitor wanting a scrubber upgrade, and assistants/clowns asking for toys. Rarely, a chemist will come by for bluespace beakers.

I would love to hand out more gear and help the station, but that is **hard to do.** As the RD, I am dealing with the AI, making sure the rest of Science and the ORM get upgrades, offering upgrades to the CMO, making sure R&D is getting done, ensuring there is something manning Robotics before assistants break in, etc. I want Science to be the department that improves **everyone**, not just themselves. @Iamgoofball is the coder who did more for that end than anyone else ever has, via the BSRPED.

In order to give people better gear, I usually have to _visit them in person_ in order to deliver it. I certainly cannot get much done if I am halfway across the station waiting to get into Engineering so I can give the CE hypercap cells and an experimental welder. Did you ever consider that Science players are so stingy simply because people do not bother to walk down and pick things up? I do not know about other RD players, but I, myself, am more than happy to help you so long as you are patient with me.

I do hope you do not plan on using the same excuse to deny Science any additional content. It is a real shame to see how the community has turned against the department. I personally feel that interdepartmental relations would be greatly improved if there were a better way of distributing items. We do not currently have any reliable system in place for it, so we resort to walking.

**Comment by hornygranny** [comment]
power creep

**Comment by WJohn** [comment]
I am okay with this PR, the odysseus mech is totally worthless right now so this, with some hard work, makes it better :+1:

**Comment by GunHog** [comment]
Could I get a merge-ready tag for this one?

**Comment by GunHog** [comment]
Conflicts resolved, code tested and functioning as expected.

@duncathan, have you changed your mind on this one? I created the PR in based on your positive reception to my proposal of making a mech version of the medical beam. Are you no longer approving of this concept?

If there are modifications needed to make this PR more acceptable, please let me know.

**Comment by KorPhaeron** [comment]
@GunHog 

If the PR is held up over "don't add it to RD" please consider making it admin only, I will make it obtainable via other means if it exists

**Comment by GunHog** [comment]
w00p w00p Thank you Dunc!

**Comment by Fox-McCloud** [comment]
And now science almost completely replaces medical at ~25 minutes.

**Comment by WJohn** [comment]
Assuming anyone actually builds odysseus mechs, which they don't. This gives people an incentive to do so, but I'm pretty sure people would rather spend their time and resources continuing on with the gygax/durand "kill everything before it can even hurt you" meta.

**Pull request merged**